rust/sdk-processor/src/steps/parquet_default_processor/size_buffer.rs

```diff
@@ -0,0 +1,260 @@
+
+        let mut upload_metadata_map = HashMap::new();
+        for (parquet_type, parquet_data) in item.data {
+            self.handle_buffer_append(
+                parquet_type,
+                parquet_data,
+                &item.metadata,
+                &mut upload_metadata_map,
+            )
+            .await?;
+        }
+
+        if !upload_metadata_map.is_empty() {
+            return Ok(Some(TransactionContext {
+                data: upload_metadata_map,
+                metadata: item.metadata, // cur_batch_metadata,
+            }));
+        }
+        Ok(None)
+    }
+
+    async fn cleanup(
+        &mut self,
+    ) -> Result<Option<Vec<TransactionContext<Self::Output>>>, ProcessorError> {
+        let mut metadata_map = HashMap::new();
+        debug!("Starting cleanup: uploading all remaining buffers.");
+        for (parquet_type, mut buffer) in self.internal_buffers.drain() {
+            if buffer.buffer_size_bytes > 0 {
+                let struct_buffer = std::mem::replace(
+                    &mut buffer.buffer,
+                    ParquetTypeStructs::default_for_type(&parquet_type),
+                );
+
+                self.buffer_uploader.handle_buffer(struct_buffer).await?;
+
+                let mut metadata = buffer.current_batch_metadata.clone();
+                metadata.total_size_in_bytes = buffer.buffer_size_bytes as u64;
+                metadata_map.insert(parquet_type, metadata);
+            }
+        }
+        self.internal_buffers.clear();
+
+        debug!("Cleanup complete: all buffers uploaded.");
+        if !metadata_map.is_empty() {
+            return Ok(Some(vec![TransactionContext {
+                data: metadata_map,
+                metadata: TransactionMetadata::default(),
+            }]));
+        }
+        Ok(None)
+    }
+}
+
+#[async_trait]
+impl<U> PollableAsyncStep for SizeBufferStep<U>
+where
+    U: Uploadable + Send + 'static + Sync,
+{
+    fn poll_interval(&self) -> Duration {
+        self.poll_interval
+    }
+
+    async fn poll(
+        &mut self,
+    ) -> Result<Option<Vec<TransactionContext<Self::Output>>>, ProcessorError> {
+        let mut metadata_map = HashMap::new();
+        debug!("Polling to check if any buffers need uploading.");
+
+        for (parquet_type, mut buffer) in self.internal_buffers.drain() {
+            if buffer.buffer_size_bytes > 0 {
+                let struct_buffer = std::mem::replace(
+                    &mut buffer.buffer,
+                    ParquetTypeStructs::default_for_type(&parquet_type),
+                );
+
+                self.buffer_uploader.handle_buffer(struct_buffer).await?;
+
+                let metadata = buffer.current_batch_metadata.clone();
+                metadata_map.insert(parquet_type, metadata);
+
+                // Reset ParquetBuffer or maybe remove?
+                buffer.buffer_size_bytes = 0;
+                buffer.current_batch_metadata = TransactionMetadata::default();
```

Comment:
Same here, when it's reset, I think it should be set to None. So the next time a buffer gets uploaded, we use the starting and ending versions from that metadata.